### DISCUSSION TITLE
Apicurio registry sql connections number

### DISCUSSION THREAD

**Discussion by Prop4et** [Apicurio/apicurio-registry#5258] [Q&A] [answered]
I created a deployment on kubernetes that uses the apicurio registry sql image to create the registry. It uses an underlying postgre db and i noticed that it opens around 22 connections towards the DB. I was wondering why is that and if it's possible to limit the number of these connections. Thank you.

**Comment by carlesarnal** [comment, marked as answer]
Sorry, I'm way behind on my GH email. Yes, it can be configured. It's set to `20` as the initial size, and by default it has a maximum of 100. You can configure it using the property `apicurio.datasource.jdbc.initial-size` or it's env var equivalent, APICURIO_DATASOURCE_JDBC_INITIAL_SIZE.

**Reply by EricWittmann** [reply]
Here are all the relevant properties you might need to control this (for Registry 3.x):

```java
    @ConfigProperty(name = "apicurio.datasource.jdbc.initial-size", defaultValue = "20")
    @Info(category = "storage", description = "Application datasource pool initial size", availableSince = "3.0.0")
    String initialSize;

    @ConfigProperty(name = "apicurio.datasource.jdbc.min-size", defaultValue = "20")
    @Info(category = "storage", description = "Application datasource pool minimum size", availableSince = "3.0.0")
    String minSize;

    @ConfigProperty(name = "apicurio.datasource.jdbc.max-size", defaultValue = "100")
    @Info(category = "storage", description = "Application datasource pool maximum size", availableSince = "3.0.0")
    String maxSize;
```
